ui/raidboss/emulator/data/network_log_converter/LineEvent.ts

```diff
@@ -1,45 +1,50 @@
 import EmulatorCommon from '../../EmulatorCommon';
+import LogRepository from './LogRepository';
 
 /**
  * Generic class to track an FFXIV log line
  */
 export default class LineEvent {
-  constructor(repo, line, parts) {
-    this.offset = 0; // To be calculated later
-
-    this.parts = parts;
-
-    this.decEvent = parseInt(this.parts[0]);
+  public offset = 0;
+  decEvent: number;
+  hexEvent: string;
+  timestamp: number;
+  networkLine: string;
+  convertedLine = '';
+  invalid = false;
+  index = 0;
+  constructor(repo: LogRepository, line: string, public parts: string[]) {
+    this.decEvent = parseInt(this.parts[0] || '');
     this.hexEvent = EmulatorCommon.zeroPad(this.decEvent.toString(16).toUpperCase());
 
-    this.timestamp = +new Date(this.parts[1]);
+    this.timestamp = +new Date(this.parts[1] || '');
 
     this.networkLine = line;
     repo.updateTimestamp(this.timestamp);
   }
 
-  convert() {
+  convert(_: LogRepository | undefined = undefined): void {
```

Comment:
It's optional. Only two subclasses use it (only the log lines that need to look up a combatant name).
0x1F - Job Gauge
0x18 - HoT/DoT tick

Reply:
Yes, but `repo` is unconditionally passed in `convertLines`, right?

Reply:
Yeah, sorry, I see where you're coming from now. Yes, it's always passed but not always used, so the question then becomes which is the better implementation practice here? Should it be on the base class but be excluded from subclass parameters? Should it be defined as an `_` argument?

Reply:
I think regardless of the answer here, the change to this file will stay the same, so I'll adjust this for the next commit and just need clarification on how to handle the `LineEvent0x**` files for the next PR at some point. I'll mark this as resolved but feel free to respond with clarification.

Reply:
So, from a TypeScript perspective, it is an error to say `convert()` in the base class and `convert(repo: LogRepository)` in the derived class, as the second function cannot be assigned to the first.  It is *technically* possible to say `convert(repo?: LogRepository)` in the base class and `convert(repo: LogRepository)` in the derived class, see [this playground](https://www.typescriptlang.org/play?strict=true#code/MYGwhgzhAEBCkFNoG8CwAoa0BmBXAdsABQAeA-AFzT64C2ARggE4CUKGWWwA9vhNyAQA6ENwDmRAAb1EVACTISAX0ksA3B2hKM29BlCQYAEWYBLAG4IAJtAQkALgnxWY8CEjSYcBYgE8qNAzMbJ6c0Dx8AsKiEpJWZpZW8si+Kuqaurr6vBD20CTQALzUCADu0CZMFtZE6egkQniERADMdQD07Q1NxHXZfHn+cIhF+Rrovo0+rXWTPbVq0J3Q9gAWpjAb0BAA1gj2wKu+QA).

However, that's a little bit sketchy.  You can assign a derived object to a variable with the type of the base class and call it with the base class prototype in an unexpected way.

What makes sense to me (both for readability, but also for correctness) is to have every function in the prototype chain have the same signature.  (Or, less strictly, to say that the base class function should list everything and the derived functions could list everything or could only list what they or anything derived from them needed?)  At any rate, that's why I was trying to argue that `repo` was required because the derived classes required it.